### PR TITLE
🐛 Don't let chart-sync overwrite production edits to charts a branch created

### PR DESCRIPTION
**Pull request by pabloarosado**
## Human summary

This PR fixes an edge case and adds a few warnings to cover uncommon cases. It also adds some unit tests and a script with further tests to check the behavior of chart sync in various scenarios (from common situations to edge cases).

---

> _Written by Claude Opus 5 — @pabloarosado at the wheel._

Stacked on [#6511 "Charts as first-class citizens in ETL, addressed by chart config UUID"](https://github.com/owid/etl/pull/6511). ETL only.

## The hole

Chart-diff already protects an existing chart from being overwritten: the approval records the target chart's version, and chart-sync re-checks it, so an edit made in production after you approved makes the approval stale and the sync is refused.

A chart **created on a branch** has no such baseline. At review time production has no such chart, so the approval is stored with no target version, and for a chart matched across environments that value is never filled in. After the merge, production's own ETL creates the chart (same config UUID, its own numeric id). If someone edits it in the production admin before chart-sync runs, the approval still matches, no conflict is raised, and the sync overwrites them silently.

## The fix

Before updating a chart matched across environments, compare what each side authored. Anything the target carries and the source does not is an edit made in the target's admin that syncing would drop: refuse, and name it. A chart the ETL has just created carries only the bootstrap `slug`, so the normal case is unaffected.

Two things are compared:

- **the authored config layer** (`charts.patchConfigId`'s row), field by field rather than as whole objects, since each side can hold a different corner of the same object (`yAxis.min` in the target, `yAxis.max` in the source);
- **tags**, because `set_tags` replaces the target's set wholesale and grapher's `setChartTags` rewrites `chart_tags` without touching `charts.updatedAt`, so a tag added in the target moves no timestamp and nothing else would notice it vanish.

A field both sides carry is deliberately not reported: the source wins there by design, and that difference is the change the reviewer approved.

`--ignore-conflicts` overrides the guard, for the case where a field deliberately deleted on the branch is indistinguishable, from the target's side, from one added there.

## Also in here

Found while testing the above, each small and each with its own commit:

- chart-sync logged nothing at all for a chart it would **create**, so a dry run reported nothing in the case you most want to know about.
- A **test that two steps can't declare the same `chart_config_id`**. Copying a chart config file copies the chart's identity, so both steps own that chart and each build overwrites the other. Found by doing it by accident: a copied config retitled a live chart on staging and took ownership of it.
- **Warnings when a config asks for something the push won't do.** `topic_tags` apply only when ETL creates the chart, and an existing chart's slug cannot be changed from ETL. Both behaviours are right; the silence was not.
- The **missing-`chart_config_id` error** now points at `etl chart-config-id`, which writes the value into the file, instead of a python one-liner and a hand-written SQL join.

## Verified end to end

`scripts/chart_sync_scenarios.py` runs against a staging server with a copy of its own database standing in for production. Eleven scenarios, 37 checks, all passing on this branch:

| | scenario | result |
|---|---|---|
| S0 | a staging build re-pushes the ETL layer unchanged | not a save, chart not listed |
| S1 | the chart is edited in production, branch never touched it | not listed |
| S2 | the branch rebuilds the chart | listed, approve/reject; with a production edit, listed with conflict |
| S3 | the chart is edited in the staging admin | listed, approve/reject |
| S4 | approved, then production is edited | approval expires, chart-sync refuses |
| S5 | **new chart, production edited before chart-sync runs** | **refused, with the lost field named; `--ignore-conflicts` gets through** |
| S6 | corner case: a deletion on staging that ordinary work never reaches | refused, and the override works |
| S7 | an indicator's numbers change, its id does not | listed **review-only**, sync writes nothing |
| S8 | an indicator's metadata changes | listed **review-only**, sync writes nothing |
| S9 | a dataset is bumped, so the chart plots a new indicator id | listed **approve/reject**, synced |
| S10 | a chart made by hand on staging | chart-sync creates it in the target |

The S5 control matters as much as S5: with production untouched the new chart still syncs, so the guard does not block the normal path. S7 to S9 assert which control chart-diff offers, since that distinction is what a data scientist actually sees.

The script is manual, not a pytest integration test: it needs Tailscale and it mutates the staging database. `make test-integration` also runs on staging servers in CI, where this would fire unattended against someone's data.

Unit tests cover the comparison function (bootstrap layer loses nothing, a target-side edit is reported, bookkeeping and shared keys ignored, nested fields, missing layers) and that `load_patch_config` reads the authored row rather than the rendered one. Each was checked to fail when the code it covers is deliberately broken.

## Not covered

- **Tags on ordinary charts.** Since `setChartTags` moves no timestamp, a tag added in production after you approve is deleted by the sync for *any* chart, not only the case fixed here. Out of scope, worth its own look.
- An edit in the target touching the same field the source changes is still overwritten, deliberately: that field is the reviewed change.
- Nothing prevents two branches approving contradictory changes to the same new chart.
- S7 and S8 stand in for one of chart-diff's two filters. The dataset's `dataEditedAt` is set for real; the git-changed-paths filter is monkeypatched, and the comment says so.
- The two new warnings have no unit test: the existing tests for that module mock below the level where they fire.

### CODE REVIEW

**Comment by pabloarosado** [comment]
@codex review

File: apps/wizard/app_pages/chart_diff/chart_diff.py [hunk 1]
```diff
@@ -1301,6 +1301,27 @@ def get_chart_diffs_from_grapher(
...
+    the target's authored layer. Any key the target's layer carries and the source's does not is
+    therefore lost. On a chart the branch created, the target's layer starts out holding only the
+    bootstrap `slug`, so anything else in it was authored in the target's own admin.
+
+    Keys present on both sides are not reported: the source wins there by design (that is the
+    change the reviewer approved, e.g. a renamed slug), and the reviewer saw it in chart-diff.
+    """
+    target_patch = target_patch or {}
+    source_patch = source_patch or {}
+    return sorted(k for k in target_patch if k not in _PATCH_BOOKKEEPING_KEYS and k not in source_patch)
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply apps/wizard/app_pages/chart_diff/chart_diff.py]: > _Written by Claude Fable 5 — @pabloarosado at the wheel._

Correct, fixed in 0bd1cb591. `patch_paths_lost_by_sync` now recurses when both sides hold a dict at the same key and reports dotted paths, so your `yAxis.min` / `yAxis.max` case returns `["yAxis.min"]` and the sync is refused. A shared *leaf* is still not reported, since there the source wins by design and that is the change the reviewer approved.

Added `test_nested_fields_are_compared_field_by_field`, covering the two-halves-of-one-object case, an identical leaf, and deeper nesting (`map.colorScale.binningStrategy`). I checked it fails when the recursion is removed.

File: apps/chart_sync/cli.py [hunk 2]
```diff
@@ -291,6 +292,32 @@ def cli(
...
+                            # `_target_updated_at_for_review`). The target's own ETL creates the
+                            # chart after the merge, and if someone edits it there before this
+                            # runs, the approval still matches and we would overwrite them.
+                            # Compare the authored layers and refuse instead.
+                            lost_keys = (
+                                patch_keys_lost_by_sync(
+                                    diff.target_chart.load_patch_config(target_session),
+                                    diff.source_chart.load_patch_config(source_session),
+                                )
+                                if cross_env_twin
```
[the remark this answers is not in the document]
Comment (pabloarosado) [reply apps/chart_sync/cli.py]: > _Written by Claude Fable 5 — @pabloarosado at the wheel._

Correct, fixed in 0bd1cb591. A target-only tag now counts as something the sync would drop, so the chart is refused and the tag named in the warning.

Worth recording what I found while checking it: grapher's `setChartTags` deletes and re-inserts `chart_tags` rows without touching `charts.updatedAt` ([db/model/Chart.ts:396](https://github.com/owid/owid-grapher/blob/etl-charts/db/model/Chart.ts#L396)). So a tag added in the target moves no timestamp, and the approval mechanism cannot see it for *any* chart, not only the cross-environment case this PR is about. This PR closes it where nothing else protects the chart; the general case is noted in the description as still open.
